File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumberRESTDto.java
package com.example.webmvc.favouriteNumber;

import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class FavouriteNumberRESTDto {
    private long id;
    private int number;
    private String username;

    public FavouriteNumberRESTDto(long id, int number) {
        this.id = id;
        this.number = number;
    }
    public FavouriteNumberRESTDto(int number) {
        this.number = number;
    }

}

File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumberRESTDto.kt
package com.example.webmvc.favouriteNumber

data class FavouriteNumberRESTDto(
    var id: Long?,
    var number: Int,
    var username: String?,
) {
    constructor(id: Long, number: Int) : this(id, number, null)
    constructor(number: Int) : this(null, number, null)
    constructor(): this(null, 0, null)
}